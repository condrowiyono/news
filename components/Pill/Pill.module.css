.pill {
  border: 1px solid #E5E9EC;
  background: transparent;
  border-radius: 32px;
  text-align: center;
  width: 100%;
  align-items: center;
  padding: 6px 16px;
  font-weight: 700;
  font-size: 12px;
  line-height: 160%;
  color: #14171A;
  cursor: pointer;

  transition: all 0.3s;
}

.pill:hover, .pill.active  {
  background: #FBD2D4;
  border: 1px solid #FBD2D4;
  color: #e2492b;
  transition: all 0.5s;
}