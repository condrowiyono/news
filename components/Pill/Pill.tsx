import React from "react";
import style from "./Pill.module.css";

interface IPill {
  children?: React.ReactNode | string;
  active?: boolean;
  onClick?: () => void;
}

const Pill = ({ children, active, onClick }: IPill) => {
  return (
    <button
      className={`${style.pill} ${active ? style.active : ""}`}
      onClick={onClick}
      data-testid="pill"
    >
      {children}
    </button>
  );
};

export default Pill;
