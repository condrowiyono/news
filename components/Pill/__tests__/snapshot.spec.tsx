import { render } from "@testing-library/react";
import Pill from "@components/Pill";

it("renders Pill unchanged", () => {
  const { container } = render(<Pill active>Title</Pill>);
  expect(container).toMatchSnapshot();
});
