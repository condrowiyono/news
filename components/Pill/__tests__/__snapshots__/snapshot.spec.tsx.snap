// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pill unchanged 1`] = `
<div>
  <button
    class="pill active"
    data-testid="pill"
  >
    Title
  </button>
</div>
`;
